hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -191,10 +191,15 @@ public static String getCommitFromCommitFile(String commitFileName) {
   }
 
   public static String getCommitTime(String fullFileName) {
-    if (isLogFile(fullFileName)) {
-      return fullFileName.split("_")[1].split("\\.")[0];
+    try {
+      if (isLogFile(fullFileName)) {
+        return fullFileName.split("_")[1].split("\\.")[0];
+      }
+      return fullFileName.split("_")[2].split("\\.")[0];
+    } catch (ArrayIndexOutOfBoundsException e) {
+      LOG.error("Found files with illegal filename! Full filename: " + fullFileName);
+      throw new HoodieException("Failed to get commit time from filename, filename is illegal: " + fullFileName, e);
```

Comment:
No need for verbose log, an exception should be enough:
```java
throw new HoodieException("Failed to get commit time from filename: " + fullFileName, e);
```

Reply:
Make sense. I've updated the code